plugins/catalog-backend/src/database/CommonDatabase.ts

```diff
@@ -330,6 +335,33 @@ export class CommonDatabase implements Database {
     }
   }
 
+  async setRelations(
+    txOpaque: unknown,
+    originatingEntityId: string,
+    relations: EntityRelationSpec[],
+  ): Promise<void> {
+    const tx = txOpaque as Knex.Transaction<any, any>;
+
+    // remove all relations that exist for the originating entity id.
+    await tx<DbEntitiesRelationsRow>('entities_relations')
+      .where({ originating_entity_id: originatingEntityId })
+      .del();
+
+    const serializeName = (e: EntityName) =>
+      `${e.kind}:${e.namespace}/${e.name}`.toLowerCase();
```

Comment:
This lowercasing thing, I've been thinking if that should ALWAYS happen (in the catalog-model functions) or not (if it should be implementation dependent like here)

Reply:
Yup, definitely. Beginning to feel I'd rather be passing around the serialized entity name too

Reply:
Hmm. Maybe EntityName should actually be a class with `matches(Entity)` (case insensitively), `toLowercaseString()`, and `from(String)`